src/AnnotationProcessor/DoctrineAnnotationProcessor.php

```diff
@@ -214,43 +219,49 @@ private function processJoinColumn(JoinColumn $join_column, PropertyInformation
      */
     private function transformType($type)
     {
-        if ($type === Type::BOOLEAN) {
+        if ($type === Types::BOOLEAN) {
             return 'boolean';
         }
 
-        if ($type === Type::SMALLINT || $type === Type::BIGINT || $type === Type::INTEGER) {
+        if ($type === Types::SMALLINT || $type === Types::BIGINT || $type === Types::INTEGER) {
             return 'integer';
         }
 
-        if ($type === Type::FLOAT) {
+        if ($type === Types::FLOAT) {
             return 'float';
         }
 
-        if ($type === Type::TEXT || $type === Type::GUID || $type === Type::STRING || $type === Type::DECIMAL) {
+        if ($type === Types::TEXT || $type === Types::GUID || $type === Types::STRING || $type === Types::DECIMAL) {
             return 'string';
         }
 
-        if ($type === Type::BLOB /* binary will be added in doctrine 2.5 */) {
+        if ($type === Types::BLOB /* binary will be added in doctrine 2.5 */) {
             return 'resource';
         }
 
         if (\in_array(
             $type,
-            [Type::DATETIME, Type::DATETIMETZ, Type::DATE, Type::TIME, self::ZEROED_DATE_TIME, self::ZEROED_DATE],
+            [
+                Types::DATETIME_MUTABLE,
+                Types::DATETIMETZ_MUTABLE,
+                Types::DATE_MUTABLE,
+                Types::TIME_MUTABLE,
+                self::ZEROED_DATE_TIME, self::ZEROED_DATE,
+            ],
             true
         )) {
             return '\\' . \DateTime::class;
         }
 
         if (\in_array(
             $type,
-            [Type::SIMPLE_ARRAY, Type::JSON_ARRAY, Type::JSON, Type::TARRAY, self::YAML_ARRAY],
+            [Types::SIMPLE_ARRAY, self::JSON_ARRAY, Types::JSON, Types::ARRAY, self::YAML_ARRAY, TYPES::JSON],
```

Comment:
Types::JSON en TYPES::JSON ? zijn dat verschillende?

en Type::TARRAY gaat naar Types::ARRAY, was TARRAY hetzelfde als ARRAY? of een typo?